pennylane/templates/layers/particle_conserving_u1.py

```diff
@@ -0,0 +1,267 @@
+            is the number of exchange gates :math:`U_{1,\mathrm{ex}}` per layer.
+        wires (Iterable or Wires): Wires that the template acts on. Accepts an iterable of numbers
+            or strings, or a Wires object.
+        init_state (array[int]): Length ``len(wires)`` occupation-number vector representing the
+            HF state. ``init_state`` is used to initialize the wires.
+
+    Raises:
+        ValueError: if inputs do not have the correct format
+
+    .. UsageDetails::
+
+        Notice that:
+
+        #. The number of wires has to be equal to the number of spin orbitals included in
+           the active space.
+
+        #. The number of trainable parameters scales linearly with the number of layers as
+           :math:`2D(N_\mathrm{qubits}-1)`.
+
+        An example of how to use this template is shown below:
+
+        .. code-block:: python
+
+            import pennylane as qml
+            from pennylane.templates import ParticleConservingU1
+            from functools import partial
+
+            # Build the electronic Hamiltonian
+            h, qubits = qml.qchem.molecular_hamiltonian("h2", "h2.xyz")
+
+            # Define the HF state
+            ref_state = qml.qchem.hf_state(electrons=2, qubits)
+
+            # Define the device
+            dev = qml.device('default.qubit', wires=qubits)
+
+            # Define the ansatz
+            ansatz = partial(ParticleConservingU1, init_state=ref_state)
+
+            # Define the cost function
+            cost_fn = qml.VQECost(ansatz, h, dev)
+
+            # Compute the expectation value of 'h'
+            layers = 2
+            params = qml.init.particle_conserving_u1_normal(layers, qubits)
+            print(cost_fn(params))
+    """
+
+    wires = Wires(wires)
+
+    layers = weights.shape[0]
+
+    if len(wires) < 2:
+        raise ValueError(
+            "This template requires the number of qubits to be >= 2; got 'len(wires)' = {}".format(
+                len(wires)
+            )
+        )
+
+    expected_shape = (layers, len(wires) - 1, 2)
+    check_shape(
+        weights,
+        expected_shape,
+        msg="'weights' must be of shape {}; got {}".format(expected_shape, get_shape(weights)),
+    )
+
+    check_type(
+        init_state,
+        [np.ndarray],
+        msg="'init_state' must be a Numpy array; got {}".format(init_state),
+    )
+    for i in init_state:
+        check_type(
+            i,
+            [int, np.int64, np.ndarray],
+            msg="Elements of 'init_state' must be integers; got {}".format(init_state),
+        )
+
+    nm_wires = [wires.subset([l, l + 1]) for l in range(0, len(wires) - 1, 2)]
+    nm_wires += [wires.subset([l, l + 1]) for l in range(1, len(wires) - 1, 2)]
+
+    qml.BasisState(init_state, wires=wires)
+
+    for l in range(layers):
+        for i, wires_ in enumerate(nm_wires):
+            u1_ex_gate(*weights[l, i], wires=wires_)
```

Comment:
You know here how many items are unpacked by the star notation, right? If so, I would recommend just indexing like `u1_ex_gate(weights[l,i,0], weights[l,i,1], wires=wires_)` for now. We found evidence elsewhere that the star notation slows down the tensorflow interface ([related](https://github.com/PennyLaneAI/pennylane/pull/858) PR). I'll keep you updated on what we find there, but should be easy to change back...

Reply:
Thanks. I will update this.